pkg/helpers/slices/slices.go

```diff
@@ -63,14 +65,31 @@ func ContainsItem[T comparable](slice []T, item T) bool {
 	return Contains(slice, IdentityFunc(item))
 }
 
-func Find[T comparable](slice []T, filterFunc func(T) bool) (T, bool) {
+func Find[T any](slice []T, filterFunc func(T) bool) (T, bool) {
+	i, found := FindIndex(slice, filterFunc)
+	if found {
+		return slice[i], true
+	}
+
+	return *new(T), false
+}
+
+func FindItem[T comparable](slice []T, item T) (T, bool) {
+	return Find(slice, IdentityFunc(item))
+}
+
+func FindIndex[T any](slice []T, filterFunc func(T) bool) (int, bool) {
```

Comment:
might be nice to share the implementation
```golang
func FindIndex[T any](slice []T, filterFunc func(T) bool) (T, int, bool) {
// for ...
}

func Find[T any](slice []T, filterFunc func(T) bool) (T, bool) {
    v, _, err := FindIndex(slice, filterFunc)
    return v, err
}
```
Or have `Find to return the index` and update the call places

Reply:
I'm leaning towards having a single function, changed